EcoReferencesPoint[] eco_ref_path
